pkgs/development/tools/protoc-gen-grpc-web/default.nix

```diff
@@ -0,0 +1,39 @@
+{ lib, stdenv, fetchFromGitHub, fetchpatch, protobuf }:
+
+stdenv.mkDerivation rec {
+  pname = "protoc-gen-grpc-web";
+  version = "1.2.1";
+
+  src = fetchFromGitHub {
+    owner = "grpc";
+    repo = "grpc-web";
+    rev = version;
+    sha256 = "sha256-NBENyc01O8NPo84z1CeZ7YvFvVGY2GSlcdxacRrQALw=";
+  };
+
+  # remove once PR merged
+  # https://github.com/grpc/grpc-web/pull/1107
+  patches = [
+    (fetchpatch {
+      name = "add-prefix";
+      url = "https://patch-diff.githubusercontent.com/raw/grpc/grpc-web/pull/1107.patch";
+      sha256 = "sha256-RNrg/ltdu9whAdfBrObiKvL8ebv22SkT/9HspHL940U=";
+    })
+  ];
+
+  postPatch = ''
+    cd ./javascript/net/grpc/web/
```

Comment:
Is there a better way to change dir or to fetchFromGithub and only keep the `./javascript/net/grpc/web/` subdir?

Reply:
Sounds like you want [`sourceRoot`](https://nixos.org/manual/nixpkgs/unstable/#var-stdenv-sourceRoot). :)

Reply:
Ah, I tried that before but I needed the prefix "source/"

e.g. `sourceRoot = "source/javascript/net/grpc/web";`